octoclient.go

```diff
@@ -27,46 +34,51 @@ type OctoResponse struct {
 
 type OctoClient struct {
 	HTTPClient *http.Client
-	BaseURL    string
+	baseURL    string
+	clientID   uuid.UUID
+}
+
+type OctoConfig struct {
+	// Other options in http.Client will be added here e.g, custom timeout
+	BaseURL  string
+	ClientID string // Use AccessToken in place of clientID
 }
 
-func New(baseUrl string) *OctoClient {
-	baseUrl = trimTrailingSlash(baseUrl)
+func New(options OctoConfig) *OctoClient {
+	baseURL := trimTrailingSlash(options.BaseURL)
+	clientID, _ := uuid.Parse(options.ClientID)
 	return &OctoClient{
 		HTTPClient: &http.Client{},
-		BaseURL:    baseUrl,
+		baseURL:    baseURL,
+		clientID:   clientID,
 	}
 }
 
 func (o *OctoClient) getHttpClient() http.Client {
 	return *o.HTTPClient
 }
 
-func (o *OctoClient) ServiceInvoke(clientID string, payload OctoPayload) (OctoResponse, error) {
+func (o *OctoClient) ServiceInvoke(payload OctoPayload) (OctoResponse, error) {
```

Comment:
`*OctoResponse` Let's return the pointer, and replace return statements with `nil, err`

Reply:
Also, add context

Reply:
make an inline func to give response body
if ctx is null 
Context.Todo()